modules/rostests/apitests/user32/ScrollBarRedraw.c

```diff
@@ -0,0 +1,720 @@
+                   "CS_VREDRAW specified, but vertical window expansion"
+                   " didn't trigger redraw, PrevColor: 0x%.8lX, Color: 0x%.8lX\n",
+                   PrevColor, Color);
+            }
+            else
+            {
+                ok(Color == PrevColor,
+                   "CS_VREDRAW not specified, but vertical window expansion"
+                   " triggered unneccessary redraw, PrevColor: 0x%.8lX, Color: 0x%.8lX\n",
+                   PrevColor, Color);
+            }
+
+            SetWindowPos(Window, HWND_TOPMOST, 0, 0, SmallWidth, SmallHeight, SWP_NOMOVE);
+
+            FsmState = FSM_STATE_BOTH_SHRUNK;
+            break;
+
+        case FSM_STATE_BOTH_SHRUNK:
+            if (HaveHRedraw || HaveVRedraw)
+            {
+                ok(Color != PrevColor,
+                   "CS_HREDRAW or CS_VREDRAW specified, but combined"
+                   " vertical/horizontal shrinkage didn't trigger redraw,"
+                   " PrevColor: 0x%.8lX, Color: 0x%.8lX\n",
+                   PrevColor, Color);
+            }
+            else
+            {
+                ok(Color == PrevColor,
+                   "Neither CS_HREDRAW nor CS_VREDRAW specified, but combined"
+                   " vertical/horizontal shrinkage triggered unneccessary redraw,"
+                   " PrevColor: 0x%.8lX, Color: 0x%.8lX\n",
+                   PrevColor, Color);
+            }
+
+            SetWindowPos(Window, HWND_TOPMOST, 0, 0, OrigWidth, OrigHeight, SWP_NOMOVE);
+
+            FsmState = FSM_STATE_BOTH_EXPANDED;
+            break;
+
+        case FSM_STATE_BOTH_EXPANDED:
+            if (HaveHRedraw || HaveVRedraw)
+            {
+                ok(Color != PrevColor,
+                   "CS_HREDRAW or CS_VREDRAW specified, but combined"
+                   " vertical/horizontal expansion didn't trigger redraw,"
+                   " PrevColor: 0x%.8lX, Color: 0x%.8lX\n",
+                   PrevColor, Color);
+            }
+            else
+            {
+                ok(Color == PrevColor,
+                   "Neither CS_HREDRAW nor CS_VREDRAW specified, but combined"
+                   " vertical/horizontal expansion triggered unneccessary redraw,"
+                   " PrevColor: 0x%.8lX, Color: 0x%.8lX\n",
+                   PrevColor, Color);
+            }
+
+            DestroyWindow(Window);
+
+            FsmState = FSM_STATE_END;
+            break;
+
+        case FSM_STATE_END:
+            break;
+    }
+
+    PrevColor = Color;
+    return 0;
+}
+
+static int OnPaint(HWND Window)
+{
+    HRGN Region;
+    HDC hdc;
+    PAINTSTRUCT ps;
+
+    hdc = BeginPaint(Window, &ps);
+    if (hdc == NULL)
+    {
+        skip("Failed to get device context\n");
+        DestroyWindow(Window);
+        return 0;
+    }
+
+    Region = CreateRectRgn(ps.rcPaint.left,
+                           ps.rcPaint.top,
+                           ps.rcPaint.right,
+                           ps.rcPaint.bottom);
+
```

Comment:
```suggestion
```

Reply:
Heh, the ReactOS team really seems to not like vertical spaces :)  I think they give the code "breathing room" and make it easier to read, but it's your project, so I'll respect your wishes.

Reply:
Thx. Vertical space can and shall be used to separate 2 things of different concern. Contrary doing a function-call and then checking its retval is dealing with the same thing semantically. So that should be kept close together ideally. A visual separation is a way to transmit information to the reader, but that works only when being applied selectively. If you try to split up everything (e.g. each letter), then you end up splitting up nothing. In general it is a nice trait for c code that can be watched at the same time, aka fitting the screen, is not too limited. A nice trait especially when dealing with allocating and freeing resources in C. It is not that we would dislike vertical space per se. We do just use it wisely to transport meaning.

Reply:
And that's completely reasonable!  Thanks for the suggestion and for your insights!